components/VM_Arm/src/main.c

```diff
@@ -715,6 +717,40 @@ static int route_irqs(vm_vcpu_t *vcpu, irq_server_t *irq_server)
     return 0;
 }
 
+/* Different from load_image which loads the image into the guest's address space. This pulls the
+ * image from the FileServer 1MB at a time and stores it into a local buffer
+ */
+static size_t get_file(const char *name, char *buf, int max_length)
+{
+    int fd;
+    size_t len = 0;
+
+    if (max_length <= 0) {
+        ZF_LOGE("Error: max_length parameter (%d) is not a positive value", max_length);
+    }
+
+    fd = open(name, 0);
+    if (fd == -1) {
+        ZF_LOGE("Error: Unable to open image \'%s\'", name);
+        return 0;
+    }
+
+    size_t offset = 0;
+    while (offset < max_length) {
+        /* Load the image 1MB at a time, until the file is read or
+         * the max_length parameter has been reached
+         */
+        len = read(fd, buf + offset, BIT(20));
```

Comment:
Need to take into account the remaining buffer size:
```suggestion
        len = read(fd, buf + offset, MIN(BIT(20), max_length - offset));
```

Reply:
Nice catch. Will update.